examples/t8code_3d_dgsem/elixir_advection_nonconforming.jl

```diff
@@ -0,0 +1,102 @@
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# Semidiscretization of the linear advection equation.
+
+advection_velocity = (0.2, -0.7, 0.5)
+equations = LinearScalarAdvectionEquation3D(advection_velocity)
+
+# Create DG solver with polynomial degree = 3 and (local) Lax-Friedrichs/Rusanov flux as surface flux.
+solver = DGSEM(polydeg=3, surface_flux=flux_lax_friedrichs)
+
+coordinates_min = (-1.0, -1.0, -1.0) # minimum coordinates (min(x), min(y), min(z))
+coordinates_max = ( 1.0,  1.0,  1.0) # maximum coordinates (max(x), max(y), max(z))
+trees_per_dimension = (1, 1, 1)
+
+mapping = Trixi.coordinates2mapping(coordinates_min, coordinates_max)
+
+# Note that it is not necessary to use mesh polydeg lower than the solver polydeg
+# on a Cartesian mesh.
+# See https://doi.org/10.1007/s10915-018-00897-9, Section 6.
+mesh = T8codeMesh(trees_per_dimension, polydeg=3,
+                  mapping=mapping,
+                  initial_refinement_level=2)
+
+function adapt_callback(forest,
```

Comment:
I agree that the API of t8code is quite cumbersome to work with. A situation I'm not satisfied with. I might be possible that the public API of t8code is subject to change in the midterm future. Hence, I refrain from investing time in designing a pretty, user-friendly interface for Julia / Trixi.jl.

Reply:
For TreeMesh we have functions like [`refine_box!`](https://github.com/trixi-framework/Trixi.jl/blob/e82ebb557a280abb87e7b6f3db9d90bcac715321/src/meshes/abstract_tree.jl#L391-L405) or [`refine_sphere!`](https://github.com/trixi-framework/Trixi.jl/blob/e82ebb557a280abb87e7b6f3db9d90bcac715321/src/meshes/abstract_tree.jl#L414-L425), which are invoked by passing [`refinement_patches`](https://github.com/trixi-framework/Trixi.jl/blob/e82ebb557a280abb87e7b6f3db9d90bcac715321/examples/tree_2d_dgsem/elixir_euler_blob_mortar.jl#L72C23-L75) to the mesh constructor. Maybe something like this could be implemented as well, which would allow to hide whatever detailed implementation is necessary by t8code.

This could/should imho be done in a subsequent PR though. Maybe create an issue for it?

Reply:
I agree with @sloede . This should be done in a separate PR. I added this to the TODO list in https://github.com/trixi-framework/Trixi.jl/issues/1556 .